Couldn't render template "mkdir /app/var/tmp: Permission denied

After `docker compose build` and `docker compose up web-server` I see this in the output:

```
web-server-1  | [2024/10/28 11:09:56] [ERROR] [/healthcheck] Couldn't render template "mkdir /app/var/tmp: Permission denied at /app/local/lib/perl5/x86_64-linux-gnu/Text/Xslate.pm line 419.
web-server-1  | "
web-server-1  | [2024/10/28 11:09:56] [ERROR] [/healthcheck] Couldn't render template "mkdir /app/var/tmp: Permission denied at /app/local/lib/perl5/x86_64-linux-gnu/Text/Xslate.pm line 419.
```
I can make it go away via:

```
diff --git a/docker-compose.yml b/docker-compose.yml
index 726e9c9d..09e323a1 100644
--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -15,8 +15,9 @@ services:
       - './:/app/'
       - 'assets:/app/root/assets/'
       - '/app/local'
+      - '/app/var'
```

In the container I then see:

```
$ find var/tmp/
var/tmp/
var/tmp/templates
var/tmp/templates/%2Fapp%2Froot
var/tmp/templates/%2Fapp%2Froot/healthcheck.txc
var/tmp/templates/%2Fapp%2Froot/base.txc
var/tmp/templates/%2Fapp%2Froot/home.txc
var/tmp/templates/%2Fapp%2Froot/inc
var/tmp/templates/%2Fapp%2Froot/inc/no_sidebar.txc
var/tmp/templates/%2Fapp%2Froot/inc/opengraph.txc
var/tmp/templates/%2Fapp%2Froot/inc/keyboard_shortcuts.txc
```

--- a/Dockerfile
+++ b/Dockerfile
@@ -46,8 +46,6 @@ RUN \
     cpm install --show-build-log-on-failure --resolver=snapshot
 EOT
 
-RUN mkdir var && chown metacpan:users var
-
 ENV PERL5LIB="/app/local/lib/perl5"
 ENV PATH="/app/local/bin:${PATH}"
 

--- a/lib/MetaCPAN/Web/View/Xslate.pm
+++ b/lib/MetaCPAN/Web/View/Xslate.pm
@@ -2,6 +2,7 @@ package MetaCPAN::Web::View::Xslate;
 use Moose;
 extends qw(Catalyst::View::Xslate);
 use File::Path           ();
+use File::Temp           ();
 use MetaCPAN::Web::Types qw( AbsPath );
 
 has '+syntax'      => ( default => 'Metakolon' );
@@ -9,8 +10,15 @@ has '+encode_body' => ( default => 0 );
 has '+preload'     => ( default => 0 );
 has '+cache'       => ( default => 0 );
 has '+cache_dir' => (
-    isa    => AbsPath,
-    coerce => 1,
+    isa     => AbsPath,
+    coerce  => 1,
+    default => sub {
+        File::Temp::tempdir(
+            TEMPLATE => 'metacpan-web-templates-XXXXXX',
+            CLEANUP  => 1,
+            TMPDIR   => 1,
+        );
+    },
 );
 has '+module' => (
     default =>

--- a/metacpan_web.conf
+++ b/metacpan_web.conf
@@ -13,7 +13,6 @@ log4perl_file   = log4perl.conf
 mark_unauthorized_releases = 0
 
 <View::Xslate>
-    cache_dir = var/tmp/templates
     cache = 1
 </View::Xslate>
 

